oat-sa/tao-core
# Feature/ADF-1056/Manage the class copy through the task queue

**PR: 1/2**

Related to: https://oat-sa.atlassian.net/browse/TR-1056

**Summary:**
Manage the client action to copy a class through the task queue component.
Also fix the server action to make it compatible with the task queue.

**What has been done:**
_(as the reformatting added some noise, please look at the commits to see the actual changes)_
- the JavaScript file `views/js/layout/actions/common.js` has been reformatted to comply with the code style. https://github.com/oat-sa/tao-core/pull/3529/commits/58e3ac00e4e384595522b8c3aaef0315e9e6a6bd
- a new `copyClassTo` action has been added, using the task queue. https://github.com/oat-sa/tao-core/pull/3529/commits/9356607f8ad460b6ee8634ac61d92a8563674c45
- the server action has been fixed as it was not using the correct parameter name, and the CSRF validation was also removed since the task queue client is not yet compatible with it. https://github.com/oat-sa/tao-core/pull/3529/commits/274d6e088b36563a41594b1461b04e15ddf4c478
- the reformatting of the FE code introduced translation loss, and it was fixed by preventing Prettier from applying to some i18n calls. https://github.com/oat-sa/tao-core/pull/3529/commits/276930359493404d2f4ec891e233b7075b4771c1

**How to test:**
- **preconditions:**
    - have several classes in the items section
    - all PR checked out (there are 2)
    - the `taoTaskQueue` extension must not be installed yet
- copy classes: it should work seamlessly
- copy a single item, it should still use the regular sync call
- install the extension `taoTaskQueue`, configure it with respect to this [readme](https://github.com/oat-sa/extension-tao-task-queue#initializing-the-queues-and-the-task-log-container)
- copy classes: it should still work seamlessly
- copy a single item, it should still use the regular sync call

Note: check also other sections (tests, deliveries, groups, test takers, ...), so far only the items are offering this new action.
